docroot/modules/custom/yukon_w3_custom/src/Plugin/Block/EditLanguageBlock.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\yukon_w3_custom\Plugin\Block;
+
+/**
+ * @file
+ * Contains \Drupal\matrics_data_upload\Plugin\Block\CertificateCheck.
+ */
+
+use Drupal\node\NodeInterface;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+
+/**
+ * Provide Committee I am Part of block.
+ *
+ * @Block(
+ *   id = "edit_language_block",
+ *   admin_label = @Translation("Edit Language Block"),
+ * )
+ */
+class EditLanguageBlock extends BlockBase implements ContainerFactoryPluginInterface {
+  /**
+   * The language manager service.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  protected $languageManager;
+
+  /**
+   * The current route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * Constructs a new BookNavigationBlock instance.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager service.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The current route match.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, LanguageManagerInterface $language_manager, RouteMatchInterface $route_match) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->languageManager = $language_manager;
+    $this->routeMatch = $route_match;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new self($configuration, $plugin_id, $plugin_definition, $container->get('language_manager'), $container->get('current_route_match'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $node = $this->routeMatch->getParameter('node');
+    $output = "";
+    if ($node instanceof NodeInterface) {
+      if ($node->hasTranslation('fr')) {
```

Comment:
This logic won't work in case FR language is an original language
<img width="1614" alt="image" src="https://github.com/ytgov/yukon-ca/assets/123640502/b400cef5-b44f-492a-9279-957185c7324b">